### PR TITLE
bypass any group claims that start with [ to fix SQL error during aut…

### PR DESCRIPTION
**Pull request by bsmithvt**
…hentication

### CODE REVIEW

File: Fabric.Identity.API/Management/UserLoginManager.cs [hunk 1]
```diff
@@ -113,7 +113,17 @@ private List<Claim> FilterClaims(IEnumerable<Claim> incomingClaims)
                 // copy the claim as-is
                 else
                 {
-                    filtered.Add(claim);
+                    if (claim.Type == FabricIdentityConstants.FabricClaimTypes.Groups)
+                    {
+                        if (string.IsNullOrEmpty(claim.Value) || !claim.Value.StartsWith("["))
```
Comment (michaelvidal24) [inline Fabric.Identity.API/Management/UserLoginManager.cs]: Do we want to add empty claims to the list?

File: Fabric.Identity.API/Management/UserLoginManager.cs [hunk 2]
```diff
@@ -113,7 +113,17 @@ private List<Claim> FilterClaims(IEnumerable<Claim> incomingClaims)
                 // copy the claim as-is
                 else
                 {
-                    filtered.Add(claim);
+                    if (claim.Type == FabricIdentityConstants.FabricClaimTypes.Groups)
```
Comment (michaelvidal24) [inline Fabric.Identity.API/Management/UserLoginManager.cs]: Should we add some of these nested if's to an else if to reduce some nesting?

File: Fabric.Identity.API/Management/UserLoginManager.cs (hunk 1 above)
Comment (bsmithvt) [reply Fabric.Identity.API/Management/UserLoginManager.cs]: We were adding them in the original version.

```
else
{
   filtered.Add(claim);
}
```

File: Fabric.Identity.API/Management/UserLoginManager.cs (hunk 2 above)
Comment (bsmithvt) [reply Fabric.Identity.API/Management/UserLoginManager.cs]: Do you mean pulling the `if (claim.Type == FabricIdentityConstants.FabricClaimTypes.Groups)` out of the `else`?

File: Fabric.Identity.API/Management/UserLoginManager.cs (hunk 1 above)
Comment (michaelvidal24) [reply Fabric.Identity.API/Management/UserLoginManager.cs]: ok - just looked weird...what claims do we get that are empty?

Comment (bsmithvt) [reply Fabric.Identity.API/Management/UserLoginManager.cs]: I don't see any in my DB that are null or empty, but the `UserClaims` table `Value` column is nullable.

File: Fabric.Identity.API/Management/UserLoginManager.cs (hunk 2 above)
Comment (michaelvidal24) [reply Fabric.Identity.API/Management/UserLoginManager.cs]: Yeah, basically combining it with the else above to make it an else if - but its not really necessary.

File: Fabric.Identity.API/Management/UserLoginManager.cs (hunk 1 above)
Comment (michaelvidal24) [reply Fabric.Identity.API/Management/UserLoginManager.cs]: ok

**Pull request merged**